### DISCUSSION TITLE
Filter one-to-many relationship

### DISCUSSION THREAD

**Discussion by flyspirit99** [Biarity/Sieve#153] [General]
I started to use Sieve a couple of days ago and love it. Now I encounter a situation and don't know how to do it by Sieve

I have a one to many relationship like below

```
public class Company
{
  public ICollection<Certificate> Certificates {get; set;}
}

public class Certificate
{
    public string Name {get; set;}
    public int CompanyId { get; set;}
    public Company Company {get; set;}
}
```

In CompanyApiController.GetCompanies() action, I need to filter company by its certificate name. Is this possible?

**Comment by MHKarami97** [comment]
Try this one:

`
[Sieve(CanFilter = true, CanSort = true)]  
public ICollection<Certificate> Certificates {get; set;}
`

`
[Sieve(CanFilter = true, CanSort = true)]  
public string Name {get; set;}`

`filters=Certificates.Name==aaaa`
